src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx

```diff
@@ -358,18 +381,72 @@ const ShortCircuitAnalysisResultTable: FunctionComponent<
         !isFetching
     );
     const rowsToShow = getRows(rows, shortCircuitAnalysisStatus);
+    const headersCsv = columns
+        .filter((column) => 'headerName' in column)
+        .map((column) => (column as { headerName?: string }).headerName || '');
+    const exportCsv = useCallback(() => {
+        downloadShortCircuitResultZippedCsv(
+            studyUuid,
+            currentNode,
+            analysisType,
+            headersCsv
+        )
+            .then((response) => {
+                debugger;
+                response.blob().then((blb: Blob) => {
+                    const url = URL.createObjectURL(blb);
+                    const link = document.createElement('a');
+                    link.href = url;
+                    link.setAttribute(
+                        'download',
+                        analysisType === ShortCircuitAnalysisType.ONE_BUS
+                            ? 'oneBus-results.zip'
+                            : 'allBuses_results.zip'
+                    );
+                    link.style.display = 'none';
+                    document.body.appendChild(link);
+                    link.click();
+                    document.body.removeChild(link);
+                });
+            })
+            .catch((error) => {
+                snackError({
+                    messageTxt: error.message,
+                    headerId: intl.formatMessage({
+                        id: 'ShortCirduitAnalysisExportCsvResultsError',
+                    }),
+                });
+            });
+    }, [studyUuid, currentNode, intl, snackError, headersCsv, analysisType]);
 
     return (
-        <Box sx={{ flexGrow: 1 }}>
-            <CustomAGGrid
-                rowData={rowsToShow}
-                defaultColDef={defaultColDef}
-                onGridReady={onGridReady}
-                getRowStyle={getRowStyle}
-                enableCellTextSelection={true}
-                columnDefs={columns}
-                overlayNoRowsTemplate={message}
-            />
+        <Box sx={styles.gridContainer}>
+            <Box sx={styles.csvExport}>
+                <Box style={{ flexGrow: 1 }}></Box>
+                <Box>
+                    <FormattedMessage id="MuiVirtualizedTable/exportCSV" />
+                </Box>
+                <Box>
+                    <IconButton
```

Comment:
A new button component arrived after merging export csv Security Analysis, see
https://github.com/gridsuite/gridstudy-app/blob/main/src/components/results/securityanalysis/security-analysis-table.tsx#L87
In the PR of Seddik, we begin using this shared button component, see 

https://github.com/gridsuite/gridstudy-app/pull/1835/files#diff-7d20d0abfbb177dd58fc68d9ebca55723faeac7d98c2c41dbd67adbc1d4cd4faR175

Reply:
refacto is done